### PR TITLE
[Feature] Support MaskedConv2d with cambricon MLU backend

### PR DESCRIPTION
**Pull request by defei-coder**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

The motivation of the PR is to support running masked_conv2d with Cambricon MLU backend.

It includes four parts:
 
    1. Add masked_conv2d_mlu_kernel.mlu src code as MLU kernel.
    2. Add masked_conv2d_mlu.cpp to support launching kernel in PyTorch.
    3. Refactor test_masked_conv2d.py and add tests/data/for_masked_conv2d folder to support testing masked_conv2d with MLU backend.
    4. Refactor common_mlu_helper.hpp to support min and max function with MLU backend.

## Modification

- MLU src code
  Add MLU src code of masked_conv2d in directory mmcv/ops/csrc/common/mlu/masked_conv2d_mlu_kernel.mlu.
- PyTorch adaptation
  Adapt masked_conv2d for PyTorch in mmcv/ops/csrc/pytorch/mlu/masked_conv2d_mlu.cpp.
- Unit test
  Support test masked_conv2d with various backends in tests/test_ops/test_masked_conv2d.py.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tests/test_ops/test_masked_conv2d.py [hunk 1]
```diff
@@ -1,15 +1,55 @@
...
 
+from mmcv.utils import IS_CUDA_AVAILABLE, IS_MLU_AVAILABLE
+
 
 class TestMaskedConv2d:
 
-    def test_masked_conv2d(self):
-        if not torch.cuda.is_available():
-            return
+    def _test_masked_conv2d(self, device):
```
Comment (grimoire) [inline tests/test_ops/test_masked_conv2d.py:11]: This test has not been used.

File: tests/test_ops/test_masked_conv2d.py [hunk 2]
```diff
@@ -1,15 +1,55 @@
...
         from mmcv.ops import MaskedConv2d
-        input = torch.randn(1, 3, 16, 16, requires_grad=True, device='cuda')
-        mask = torch.randn(1, 16, 16, requires_grad=True, device='cuda')
-        conv = MaskedConv2d(3, 3, 3).cuda()
+        input = torch.randn(1, 3, 16, 16, requires_grad=True, device=device)
+        mask = torch.randn(1, 16, 16, requires_grad=True, device=device)
+        if device == 'cuda':
```
Comment (grimoire) [inline tests/test_ops/test_masked_conv2d.py:15]: Could we use `conv = MaskedConv2d(3, 3, 3).to(device)`?

Comment (defei-coder) [reply tests/test_ops/test_masked_conv2d.py]: Yep, your suggestion is very good, I have revised it.

File: tests/test_ops/test_masked_conv2d.py (hunk 1 above)
Comment (defei-coder) [reply tests/test_ops/test_masked_conv2d.py]: Deleted

**Review by grimoire — APPROVED** [review]
LGTM

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**